lib/instrumentation/index.js

```diff
@@ -76,11 +103,20 @@ Instrumentation.prototype.start = function () {
     require('./patch-async')(this)
   }
 
+  this._startHook()
+}
+
+Instrumentation.prototype._startHook = function () {
+  if (this._hook) {
+    this._hook.unhook()
```

Comment:
What's the reason for this check?

Reply:
Ah, right...I started working on a way to rebuild the hook rather than relying on the edge-case of `require-in-the-middle` _just working_ when we push new elements into the array of module paths given to it. Forgot about that. 😅

Reply:
fixed